### PR TITLE
Add timer for total time step and reformat timer output

### PR DESCRIPTION
**Pull request by jrood-nrel**
This looks like:
```
Step: 1
  Tioga: 0.002000 0.002000 0.002000
  Total: 0.002000
Nalu-Wind step: 1
  Pre: 0.000000 0.000000 0.000000
  PreConn: 0.000000 0.000500 0.001000
  PostConn: 0.001000 0.001000 0.001000
  Register: 0.000000 0.000000 0.000000
  Update: 0.000000 0.000000 0.000000
  Solve: 0.158000 0.158000 0.158000
  Post: 0.003000 0.003000 0.003000
  Total: 0.162500
AMR-Wind step: 1
  Pre: 0.000000 0.000000 0.000000
  PreConn: 0.000000 0.000500 0.001000
  PostConn: 0.005000 0.005500 0.006000
  Register: 0.006000 0.006000 0.006000
  Update: 0.006000 0.006500 0.007000
  Solve: 4.898000 4.898000 4.898000
  Post: 0.000000 0.000000 0.000000
  Total: 4.916500
Total: 5.086000

Nalu-Wind Memory Usage min: -1 avg: -1 max: -1 total: -2
AMR-Wind Memory Usage min: -1 avg: -1 max: -1 total: -2

Step: 2
```

### CODE REVIEW

**Review by ashesh2512 — APPROVED** [review]

**Comment by jrood-nrel** [comment]
Actually maybe I need to do a reduction to make sure I account for every rank? Or are there MPI barriers that make that overkill?

**Review by PaulMullowney — APPROVED** [review]

File: src/OversetSimulation.cpp [hunk 1]
```diff
@@ -127,9 +127,11 @@ void OversetSimulation::run_timesteps(const int add_pic_its, const int nsteps)
         std::to_string(tstart));
 
     for (int nt = tstart; nt < tend; ++nt) {
+        using ClockT = std::chrono::steady_clock;
```
Comment (marchdf) [inline src/OversetSimulation.cpp:130]: We could use the `tick` `tock` functionality for this and the stuff below? Or is there a reason not to?

**Comment by marchdf** [comment]
As for the reductions... I think the barriers take care of it.

**Comment by marchdf** [comment]
Looking at other codes... 

Nalu-wind does: `WallClockTime: 2 Pre: 0.000266075 NLI: 0.639814 Post: 0.00320911 Total: 0.643289` 
Which is kinda what we had before and is not very human readable

amrex does this at the end of a run: 
```
-----------------------------------------------------------------------------------------------
Name                                            NCalls  Excl. Min  Excl. Avg  Excl. Max   Max %
-----------------------------------------------------------------------------------------------
EB2::GShopLevel()-fine                               1      236.4      236.4      236.5  76.32%
FabArray::ParallelCopy_finish()                  33778      22.88      25.41      27.51   8.88%
VisMF::Write(FabArray)                              16      10.69      10.89      11.91   3.85%
DistributionMapping::LeastUsedCPUs()              1109      3.992      5.264      6.484   2.09%
FillBoundary_finish()                             1052      2.291      3.647      4.808   1.55%
PeleC::getMOLSrcTerm()                            2904       1.79      2.772      4.472   1.44%
TagBoxArray::collate()                             738      2.678      2.984      3.437   1.11%
PeleC::initialize_eb2_structs()                    397     0.6363      1.291      2.062   0.67%
```

Maybe we do:
```
----------------------------------------------------------------------
Name                             step        min        avg        max
----------------------------------------------------------------------
Tioga::Connectivity                 1      236.4      236.4      236.5
Nalu-Wind::Preconn                  1      22.88      25.41      27.51
Nalu-Wind::Post                     1      10.69      10.89      11.91
[...]
AMR-Wind::Preconn                   1      3.992      5.264      6.484
[...]
```

**Comment by PaulMullowney** [comment]
Nalu-Wind also has the per equation timing output. It would be nice to have that also.

**Comment by jrood-nrel** [comment]
I think Marc's last table is probably what I would prefer in the end. I'm not sure I'm prepared to implement that at the moment, but that would be a good direction. However, I don't think this output is too difficult to parse:
```
% egrep -A 8 "AMR-Wind step" out.txt | grep Total | awk '{print $2}'
4.672500
4.803500
4.213500
4.129000
3.631500
3.895000
3.821500
3.866000
4.122500
4.044000
```

File: src/OversetSimulation.cpp (hunk 1 above)
Comment (jrood-nrel) [reply src/OversetSimulation.cpp:130]: Yeah I will look at it again.

**Comment by jrood-nrel** [comment]
I think to get to that table will require a refactoring of how the timer output is currently done. I've been trying to work though it.

**Comment by jrood-nrel** [comment]
Actually, I can get it part of the way to the table and then will have to leave the rest for a refactor.

**Comment by marchdf** [comment]
Sounds good. Also it's fine it you have other things to do, we can do it later too.

**Comment by jrood-nrel** [comment]
Ok it looks like this. I think this is as far as I'm going to go in this PR.
```
Exawind step: 9
  Exawind:    4.9210  4.9210  4.9210
  Total:      4.9210
  Tioga:      0.0010  0.0015  0.0020
  Total:      0.0015
Nalu-Wind step: 9
  Pre:        0.0000  0.0000  0.0000
  PreConn:    0.0000  0.0000  0.0000
  PostConn:   0.0020  0.0020  0.0020
  Register:   0.0000  0.0000  0.0000
  Update:     0.0000  0.0000  0.0000
  Solve:      0.1920  0.1920  0.1920
  Post:       0.0040  0.0040  0.0040
  Total:      0.1980
AMR-Wind step: 9
  Pre:        0.0000  0.0000  0.0000
  PreConn:    0.0010  0.0010  0.0010
  PostConn:   0.0080  0.0080  0.0080
  Register:   0.0080  0.0080  0.0080
  Update:     0.0050  0.0055  0.0060
  Solve:      4.7050  4.7050  4.7050
  Post:       0.0000  0.0000  0.0000
  Total:      4.7275
Nalu-Wind memory -- min: -1 avg: -1 max: -1 total: -2
AMR-Wind memory -- min: -1 avg: -1 max: -1 total: -2
```

**Comment by ashesh2512** [comment]
@jrood-nrel For an output like the below, what would be the appropriate way to grep nalu-wind time totals for each instance ?

Exawind step: 480512
  Exawind:   16.8650 16.8670 16.8670
Nalu-Wind step: 12
  Pre:        0.0000  0.0000  0.0000
  PreConn:    0.0000  0.0000  0.0000
  PostConn:   0.0010  0.0010  0.0010
  Register:   0.0000  0.0000  0.0000
  Update:     0.0000  0.0000  0.0000
  Solve:      2.6320  2.7641  2.7940
  Post:       0.0040  0.0137  0.0190
  Total:     16.8670
Nalu-Wind step: 12
  Pre:        0.0000  0.0000  0.0000
  PreConn:    0.0000  0.0001  0.0010
  PostConn:   0.0010  0.0010  0.0010
  Register:   0.0000  0.0000  0.0000
  Update:     0.0000  0.0000  0.0000
  Solve:      2.8600  2.9203  2.9470
  Post:       0.0040  0.0114  0.0160
Nalu-Wind step: 12
  Pre:        0.0000  0.0000  0.0000
  PreConn:    0.0000  0.0000  0.0000
  PostConn:   0.0000  0.0009  0.0010
  Register:   0.0000  0.0000  0.0000
  Update:     0.0000  0.0000  0.0000
  Solve:      2.6180  2.7242  2.7510
  Post:       0.0040  0.0132  0.0190
Nalu-Wind step: 12
  Pre:        0.0000  0.0000  0.0000
  PreConn:    0.0000  0.0000  0.0000
  PostConn:   0.0000  0.0009  0.0010
  Register:   0.0000  0.0000  0.0000
  Update:     0.0000  0.0000  0.0000
  Solve:      2.6180  2.7242  2.7510
  Post:       0.0040  0.0132  0.0190
Nalu-Wind step: 12
  Pre:        0.0000  0.0000  0.0000
  PreConn:    0.0000  0.0000  0.0000
  PostConn:   0.0000  0.0009  0.0010
  Register:   0.0000  0.0000  0.0000
  Update:     0.0000  0.0000  0.0000
  Solve:      2.7180  2.7949  2.8200
  Post:       0.0040  0.0126  0.0170
  Total:      2.7788
  Tioga:      0.0050  0.2868  0.9760
  Total:      2.9328
  Total:      2.7383
  Total:      2.8084
Nalu-Wind memory -- min: 465 avg: 511 max: 561 total: 257936
  Total:      0.2868
Nalu-Wind memory -- min: 458 avg: 512 max: 580 total: 258257
Nalu-Wind memory -- min: 459 avg: 512 max: 561 total: 258308
Nalu-Wind memory -- min: 462 avg: 512 max: 599 total: 258291
AMR-Wind step: 480512
  Pre:        0.0150  0.0178  0.0430
  PreConn:    0.0010  0.0011  0.0070
  PostConn:   0.6770  0.6790  0.6820
  Register:   0.0820  0.1063  0.1110
  Update:     0.0760  0.3252  0.3340
  Solve:      8.6210  8.6301  8.6350
  Post:       0.0010  0.0044  0.0120
  Total:      9.7638
AMR-Wind memory -- min: 730 avg: 823 max: 926 total: 3319655

**Comment by jrood-nrel** [comment]
Hmm, I guess I haven't run with multiple nalu instances with this. We probably need some barriers or flushing to get the output ordered, or I guess would could just over-label each timer.

**Pull request merged**